src/twisted/test/test_defer.py

```diff
@@ -1712,6 +1729,14 @@ def ic(d: object) -> Generator[Any, Any, None]:
         self.assertEqual("test_inlineCallbacksTracebacks", tb[0][2])
         self.assertEqual("f.raiseException()", tb[0][3])
 
+    def test_fromCoroutine(self) -> None:
```

Comment:
Can you please add a docstring explaining the scope/purpose of this test?

Thanks

Maybe 
```suggestion
    def test_fromCoroutine(self) -> None:
        """
        It can discover the return type of a first level coroutine.
        """
```

Now ... what `first-level` means :)
I can see that the coroutines are not chained ... not sure if the behaviour is different if you have a coroutine returning another coroutine.